src/Mod/Sketcher/App/SketchObjectPyImp.cpp

```diff
@@ -1043,16 +1043,16 @@ PyObject* SketchObjectPy::getAxis(PyObject *args)
 PyObject* SketchObjectPy::fillet(PyObject *args)
 {
     PyObject *pcObj1, *pcObj2;
-    int geoId1, geoId2, posId1, trim=1;
+    int geoId1, geoId2, posId1, trim=1, createCorner=0;

```

Comment:
Could we use a boolean instead?

Look here for an example of an optional boolean parameter:
https://github.com/FreeCAD/FreeCAD/blob/eb6167ff89bc2b287c83d726dfcd52b775d1757e/src/Mod/Sketcher/App/SketchObjectPyImp.cpp#L488

Reply:
Done.  Looks like there's a 'p' type for bools now since python 3.3, which is a bit simpler, so I used that.

Reply:
I am not sure that is ok, for the reason of maintaining legacy support. I think we are still supporting Python2 ATM. We better ask Werner.

@wwmayer 
using p for bools is ok for current code to be merged into master at this time?

Reply:
@emmanuelobrien

I have just realised that you have changed the preexisting "trim" parameter from i to p too. 

That changes the prototype of the function as currently used for scripts and macros (before taking an int, now taking a boolean). This we should not do, as we would break those existing scripts (Yes, it would have been great to have added boolean at the time, but we need to support int at this moment). 

I admit I did not realise of the preexisting trim parameter being an int used as a boolean in my previous review. Sorry about that.

So we could:
1. Change back the new boolean parameter to int => I do not quite like it, but it is consistent with the preexisting one.
2. Leave the preexisting one as int and add the new one as boolean => not very consistent
3. Provide support for both the trim parameter being an int or a boolean and add the new one as boolean => Probably the best option, we provide a new API that admits trim as boolean and enforces the new parameter as boolean, while at the same time, we support the legacy int for preexisting scripts.

What do you think?

Reply:
> using p for bools is ok for current code to be merged into master at this time?

We should still properly handle this for Py2, IMO.

Reply:
Thanks for confirming Werner.

@emmanuelobrien

Do you feel you could implement (3) following the example I linked so that it is Py2 compatible?